hoomd/mpcd/ParticleData.cc

```diff
@@ -850,9 +850,13 @@ unsigned int mpcd::ParticleData::getTag(unsigned int idx) const
 /*!
  * \param N Allocate space for \a N additional virtualq particles in the particle data arrays
  */
-void mpcd::ParticleData::addVirtualParticles(unsigned int N)
+unsigned int mpcd::ParticleData::addVirtualParticles(unsigned int N)
     {
-    if (N == 0) return;
+    const unsigned int first_idx = getN() + getNVirtual();
```

Comment:
Generally prefer to use a class's member variables internally:
```suggestion
    const unsigned int first_idx = m_N + m_N_virtual;
```